### PR TITLE
fix: deduplicate snapshot rumors

### PR DESCRIPTION
**Pull request by effects-ai**

### CODE REVIEW

File: modules/currency-l0/src/test/scala/io/constellationnetwork/currency/l0/snapshot/SnapshotEventsPublisherDaemonSuite.scala [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    (Supervisor[IO], Hasher[IO], SecurityProvider[IO], KeyPair)
+
+  override def sharedResource: Resource[IO, Res] =
+    for {
+      s <- Supervisor[IO](await = true)
+      implicit0(js: JsonSerializer[IO]) <- JsonSerializer.forSync[IO].toResource
+      implicit0(sp: SecurityProvider[IO]) <- SecurityProvider.forAsync[IO]
+      h = Hasher.forJson[IO]
+      m = NoOpMetrics.make
+      e = {
```
Comment (marcinwadon) [inline modules/currency-l0/src/test/scala/io/constellationnetwork/currency/l0/snapshot/SnapshotEventsPublisherDaemonSuite.scala:46]: Leftover?

**Review by marcinwadon — CHANGES_REQUESTED** [review]
I see you've closed your previous PR and opened a new one. There were some comments left, please take a look.

File: modules/currency-l0/src/test/scala/io/constellationnetwork/currency/l0/snapshot/SnapshotEventsPublisherDaemonSuite.scala [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  override type Res =
+    (Supervisor[IO], Hasher[IO], SecurityProvider[IO], KeyPair)
+
+  override def sharedResource: Resource[IO, Res] =
+    for {
+      s <- Supervisor[IO](await = true)
+      implicit0(js: JsonSerializer[IO]) <- JsonSerializer.forSync[IO].toResource
+      implicit0(sp: SecurityProvider[IO]) <- SecurityProvider.forAsync[IO]
+      h = Hasher.forJson[IO]
+      kp <- KeyPairGenerator.makeKeyPair[IO].toResource
```
Comment (marcinwadon) [inline modules/currency-l0/src/test/scala/io/constellationnetwork/currency/l0/snapshot/SnapshotEventsPublisherDaemonSuite.scala:45]: `KeyPairGenerator` does not need to be a resource

File: modules/currency-l0/src/test/scala/io/constellationnetwork/currency/l0/snapshot/SnapshotEventsPublisherDaemonSuite.scala [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      gossip <- Gossip.make[IO](rumorQueue, selfId, Generation(PosLong(1)), keyPair)
+      _ <- SnapshotEventsPublisherDaemon.make(gossip, events, consensusStorage).spawn.start
+      _ <- waitUntilQueueSize(l1OutputQueue, 0)
+      _ <- waitUntilQueueSize(rumorQueue, 1)
+      rumorQueueSize <- rumorQueue.size
+      maybeRumor <- rumorQueue.tryTake
+      maybeEitherBlockEvent = maybeRumor.map { rumor =>
+        val cur = rumor.signed.value.content.hcursor
+        cur.downField("value").get[BlockEvent]("BlockEvent")
+      }
```
Comment (marcinwadon) [inline modules/currency-l0/src/test/scala/io/constellationnetwork/currency/l0/snapshot/SnapshotEventsPublisherDaemonSuite.scala:120]: what is the Rumor type? do you have to manually play with json to extract the type of the rumor?

File: modules/node-shared/src/test/scala/io/constellationnetwork/node/shared/infrastructure/metrics/NoOpMetrics.scala [hunk 4]
```diff
@@ -0,0 +1,66 @@
+package io.constellationnetwork.node.shared.infrastructure.metrics
+
+import cats.effect.IO
+
+import scala.concurrent.duration.FiniteDuration
+
+import io.constellationnetwork.node.shared.infrastructure.metrics.Metrics.{MetricKey, TagSeq}
+
+object NoOpMetrics {
```
Comment (kpudlik) [inline modules/node-shared/src/test/scala/io/constellationnetwork/node/shared/infrastructure/metrics/NoOpMetrics.scala:9]: I wish we had a quicker way for stubbing in tests. Unless we already have 🤔

**Review by kpudlik — APPROVED** [review]

**Review by marcinwadon — APPROVED** [review]

**Pull request merged**